app/models/king.rb

```diff
@@ -1,2 +1,11 @@
 class King < Piece
+	def valid_move?(new_rank, new_file)
+		rank_diff = (new_rank - rank).abs
+		file_diff = (new_file - file).abs
+		if (rank_diff <= 1) && (file_diff <= 1) && !is_move_obstructed?(new_rank, new_file) && !is_move_malformed?(new_rank, new_file)
```

Comment:
oh...i understand now. it is a great point that i didn't catch it.
I will modify it. Thanks.